HW_1/hw_1.py

```diff
@@ -0,0 +1,466 @@
+# -*- coding: utf-8 -*-
+"""HW_1.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1h83zgHyzmc6OacKk0k-aB4qEE969xzMD
+
+# Задание 1 (5 баллов)
+
+Напишите классы **Chat**, **Message** и **User**. Они должны соответствовать следующим требованиям:
+
+**Chat**:
++ Должен иметь атрибут `chat_history`, где будут храниться все сообщения (`Message`) в обратном хронологическом порядке (сначала новые, затем старые)
++ Должен иметь метод `show_last_message`, выводящий на экран информацию о последнем сообщении
++ Должен иметь метод `get_history_from_time_period`, который принимает два опциональных аргумента (даты с которой и по какую мы ищем сообщения и выдаём их). Метод также должен возвращать объект типа `Chat`
++ Должен иметь метод `show_chat`, выводящий на экран все сообщения (каждое сообщение в таком же виде как и `show_last_message`, но с разделителем между ними)
++ Должен иметь метод `recieve`, который будет принимать сообщение и добавлять его в чат
+
+**Message**:
++ Должен иметь три обязательных атрибута
+    + `text` - текст сообщения
+    + `datetime` - дата и время сообщения (встроенный модуль datetime вам в помощь). Важно! Это должна быть не дата создания сообщения, а дата его попадания в чат! 
+    + `user` - информация о пользователе, который оставил сообщение (какой тип данных использовать здесь, разберётесь сами)
++ Должен иметь метод `show`, который печатает или возвращает информацию о сообщении с необходимой информацией (дата, время, юзер, текст)
++ Должен иметь метод `send`, который будет отправлять сообщение в чат
+
+**User**:
++ Класс с информацией о юзере, наполнение для этого класса придумайте сами
+
+Напишите несколько примеров использования кода, которое показывает взаимодействие между объектами.
+
+В тексте задания намерено не указано, какие аргументы должны принимать методы, пускай вам в этом поможет здравый смысл)
+
+В этом задании не стоит флексить всякими продвинутыми штуками, для этого есть последующие
+
+В этом задании можно использовать только модуль `datetime`
+"""
+
+import datetime
+
+class Chat:
+    
+    def __init__(self, chat_history=[]):
```

Comment:
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/